Runtime/Conditions/PositionalConditions/ObjectInRangeCondition.cs

```diff
@@ -26,13 +28,40 @@ public class EntityData : IObjectInTargetData
             [DataMember]
             [DisplayName("Object")]
             public SceneObjectReference Target { get; set; }
-
+            
             /// <summary>
             /// The object to measure distance from.
             /// </summary>
             [DataMember]
             [DisplayName("Reference object")]
-            public SceneObjectReference DistanceDetector { get; set; }
+            public ScenePropertyReference<TransformInRangeDetectorProperty> ReferenceProperty { get; set; }
+            
+            /// <summary>
+            /// The object to measure distance from.
+            /// </summary>
+            [DataMember]
+            [HideInTrainingInspector]
+            [Obsolete("")]
```

Comment:
```suggestion
            [Obsolete("Use 'ReferenceProperty' instead.")]
```

Reply:
it should actually take of the reference but still be broken due to missing property, which I could autofix, if wanted. But as far as I  have tested it, the validation system will also warn here.